packages/wrangler/src/tail/index.ts

```diff
@@ -17,225 +17,233 @@ import {
 	prettyPrintLogs,
 	translateCLICommandToFilterMessage,
 } from "./createTail";
-import type {
-	CommonYargsArgv,
-	StrictYargsOptionsToInterface,
-} from "../yargs-types";
 import type { TailCLIFilters } from "./createTail";
 import type WebSocket from "ws";
 
-export function tailOptions(yargs: CommonYargsArgv) {
-	return yargs
-		.positional("worker", {
+defineCommand({
+	command: "wrangler tail",
+	metadata: {
+		description: "🦚 Start a log tailing session for a Worker",
+		owner: "Workers: Authoring and Testing",
+		status: "stable",
+	},
+	positionalArgs: ["worker"],
+	args: {
+		worker: {
 			describe: "Name or route of the worker to tail",
 			type: "string",
-		})
-		.option("format", {
-			default: process.stdout.isTTY ? "pretty" : "json",
+		},
+		format: {
+			default: "pretty",
 			choices: ["json", "pretty"],
 			describe: "The format of log entries",
-		})
-		.option("status", {
+		},
+		status: {
 			choices: ["ok", "error", "canceled"],
 			describe: "Filter by invocation status",
 			array: true,
-		})
-		.option("header", {
+		},
+		header: {
 			type: "string",
 			requiresArg: true,
 			describe: "Filter by HTTP header",
-		})
-		.option("method", {
+		},
+		method: {
 			type: "string",
 			requiresArg: true,
 			describe: "Filter by HTTP method",
 			array: true,
-		})
-		.option("sampling-rate", {
+		},
+		"sampling-rate": {
 			type: "number",
 			describe: "Adds a percentage of requests to log sampling rate",
-		})
-		.option("search", {
+		},
+		search: {
 			type: "string",
 			requiresArg: true,
 			describe: "Filter by a text match in console.log messages",
-		})
-		.option("ip", {
+		},
+		ip: {
 			type: "string",
 			requiresArg: true,
 			describe:
 				'Filter by the IP address the request originates from. Use "self" to filter for your own IP',
 			array: true,
-		})
-		.option("version-id", {
+		},
+		"version-id": {
 			type: "string",
 			requiresArg: true,
 			describe: "Filter by Worker version",
-		})
-		.option("debug", {
+		},
+		debug: {
 			type: "boolean",
 			hidden: true,
 			default: false,
 			describe:
 				"If a log would have been filtered out, send it through anyway alongside the filter which would have blocked it.",
-		})
-		.option("legacy-env", {
+		},
+		"legacy-env": {
 			type: "boolean",
 			describe: "Use legacy environments",
 			hidden: true,
+		},
+	},
+	behaviour: {
+		printBanner: false,
+	},
+	async handler(args, { config }) {
+		if (!process.stdout.isTTY) {
+			args.format = "json";
+		}
+		if (args.format === "pretty") {
+			await printWranglerBanner();
+		}
```

Comment:
Might be worth standardizing this behavior in the future?

Reply:
as in make `format=pretty` a global arg?

Reply:
I thought this is about the wrangler type command and I wasn't sure why we wanna hide the banner. But I guess it makes sense only in this particular case 🤔